src/main/java/org/truffleruby/interop/ToJavaStringNode.java

```diff
@@ -40,8 +46,24 @@ String stringCached(DynamicObject value,
     }
 
     @Specialization(guards = "isRubyString(value)", replaces = "stringCached")
-    public String stringUncached(DynamicObject value) {
-        return StringOperations.getString(value);
+    public String stringUncached(DynamicObject value,
+            @Cached("createBinaryProfile()") ConditionProfile asciiOnlyProfile,
+            @Cached("create()") RopeNodes.BytesNode bytesNode) {
+        final Rope rope = StringOperations.rope(value);
+        final byte[] bytes = bytesNode.execute(rope);
+
+        if (asciiOnlyProfile.profile(rope.isAsciiOnly())) {
+            return RopeOperations.decodeAscii(bytes, 0, bytes.length);
```

Comment:
`StringOperations.getStrin()` already dispatches to `decodeAscii()`, but with an unprofiled branch.
So this does the same thing but profiles the branch, right?

If this kind of logic is done in multiple places, I think we should have a Node for it to abstract the details.

Reply:
That's correct. I'm not sure what you mean by having a node for it. Isn't that what I'm doing here (i.e., adding a node for it)?

Reply:
I meant this logic alone (efficient Ruby String to Java String) should have its own node if it's used on the fast path somewhere else. But otherwise here is fine, it's just not ideal to leak implementation details here but well this is a "ToJavaStringNode" after all.